src/dune/context.ml

```diff
@@ -746,6 +746,10 @@ module DB = struct
         ~output:(Simple (module T))
         ~visibility:Hidden Sync
         (fun name ->
+          (* CR-someday amokhov: Here we assume that [get] is called after the
+             asynchronously running function [Create.memo] has completed. It
+             would be better to statically guarantee the completion, e.g. by
+             moving this code into a fiber. *)
```

Comment:
This was ruled out because it would require any functions which need the context to be inside the fiber. Me and @diml decided that it was too large and possibly undesirable refactoring

Reply:
@rgrinberg Thanks for providing more context! I think it's best to expand the comment, which I've done in #2998.